.ci/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,73 @@
+void cancelPrevious() {
+    stage('Cancelling previous') {
+        def buildNumber = env.BUILD_NUMBER as int
+        if (buildNumber > 1) milestone(buildNumber - 1)
+        milestone(buildNumber)
+    }
+}
+
+private Closure runTests(String nodeName) {
+    def ret = {
+        node(nodeName) {
+            stage("${nodeName}") {
+                try {
+                    def scmVars = checkout scm
+                    if (nodeName == 'Linux') {
+                        def dockerImage = 'conanio/ci-functional'
+                        docker.image(dockerImage).inside('--entrypoint=') {
+                            def sourcedir = '/home/conan/'
+                            def pyenvdir = "${sourcedir}.pyenv"
+                            sh(script: "cp -R ./ ${sourcedir}")
+                            sh(script: "chown -R conan ${sourcedir}")
+                            def command = "su - conan -c \"" \
+                                        + "export PYENV_ROOT=${pyenvdir} && " \
+                                        + "export PATH=\"${pyenvdir}/bin:\$PATH\" && " \
+                                        + "pyenv local \$PY36 \$PY38 \$PY39 && " \
```

Comment:
Better to _enable_ only one, right? What is the purpose of adding several versions here?

Reply:
This is because tox did not find the different python versions without it.

Reply:
.... although now we are only using py36 in `tox` for all the jobs...